micro-rdk/src/common/data_manager.rs

```diff
@@ -658,41 +669,57 @@ impl DataCollectAndSyncTask {
                         err
                     )
                 });
-            if let Ok(mut readings) = readings {
-                match time_correct_reading(self.robot_start_time, &mut readings) {
-                    Ok(_) => {
-                        let upload_request = DataCaptureUploadRequest {
-                            metadata: Some(UploadMetadata {
-                                part_id: self.part_id.clone(),
-                                component_type: collector_key.component_type.clone(),
-                                r#type: DataType::TabularSensor.into(),
-                                component_name: collector_key.r_name.clone(),
-                                method_name: collector_key.method.to_string(),
-                                ..Default::default()
-                            }),
-                            sensor_contents: vec![readings],
-                        };
-                        match app_client.upload_data(upload_request).await {
-                            Ok(_) => {
-                                #[cfg(feature = "data-upload-hook-unstable")]
-                                unsafe {
-                                    micro_rdk_data_manager_post_upload_hook();
-                                }
-                            }
-                            // if we can't upload, don't try again until after the next sleep
-                            Err(err) => {
-                                log::error!("error uploading data, failed to upload {:?} readings on this attempt: {}", collectors_len - idx, err);
+            if let Ok(readings) = readings {
+                let robot_start_time = self.robot_start_time;
+                let readings: Vec<SensorData> = readings
+                    .into_iter()
+                    .filter_map(|mut readings| {
+                        if is_cached_readings {
+                            Some(readings)
+                        } else if let Err(err) =
+                            time_correct_reading(robot_start_time, &mut readings)
+                        {
+                            log::error!(
+                                "unable to time correct reading for collector {:?}, {:?}",
+                                collector_key,
+                                err
+                            );
+                            None
+                        } else {
+                            Some(readings)
+                        }
+                    })
+                    .collect();
+                if !readings.is_empty() {
+                    let upload_request = DataCaptureUploadRequest {
+                        metadata: Some(UploadMetadata {
```

Comment:
if we hit the maximum request size how is this handled? fine is its undefined just curious